internal/buildjob/buildjob_test.go

```diff
@@ -0,0 +1,235 @@
+	}
+
+	inlineUsername := "marge"
+	inlinePassword := "simpson"
+	inlineHost := "inline-test.com"
+
+	inlineRegistryConfig := v1alpha1.Registry{
+		Server: inlineHost,
+		BasicAuth: v1alpha1.BasicAuthConfig{
+			Username: inlineUsername,
+			Password: inlinePassword,
+		},
+	}
+
+	inlineRegistry := config.Registry{
+		Host:     inlineHost,
+		Username: inlineUsername,
+		Password: inlinePassword,
+		NonSSL:   false,
+	}
+
+	secretUsername1 := "johndoe"
+	secretPassword1 := "anothersecurepassword"
+	secretHost1 := "secret-test.com"
+
+	secretRegistry1 := config.Registry{
+		Host:     secretHost1,
+		Username: secretUsername1,
+		Password: secretPassword1,
+		NonSSL:   false,
+	}
+
+	secretUsername2 := "janedoe"
+	secretPassword2 := "yetanothersecurepassword"
+	secretHost2 := inlineHost // intentional conflict with inline auth host
+
+	secretRegistry2 := config.Registry{
+		Host:     secretHost2,
+		Username: secretUsername2,
+		Password: secretPassword2,
+		NonSSL:   false,
+	}
+
+	singleRegistrySecretName := "SecretWithSingleRegistry"
+	singleRegistrySecretNamespace := "SomeNamespace"
+	singleRegistrySecretContent := []byte(
+		fmt.Sprintf(`{"auths":{"%v":{"username":"%v","password":"%v"}}}`, secretHost1, secretUsername1, secretPassword1))
+
+	singleRegistrySecretRegistryConfig := v1alpha1.Registry{
+		Server: secretHost1,
+		BasicAuth: v1alpha1.BasicAuthConfig{
+			SecretName:      singleRegistrySecretName,
+			SecretNamespace: singleRegistrySecretNamespace,
+		},
+	}
+
+	twoRegistriesSecretName := "SecretWithTwoRegistries"
+	twoRegistriesSecretNamespace := "SomeNamespace"
+	twoRegistriesSecretContent := []byte(
+		fmt.Sprintf(`{"auths":{"%v":{"username":"%v","password":"%v"}, "%v":{"username":"%v","password":"%v"}}}`,
+			secretHost1, secretUsername1, secretPassword1,
+			secretHost2, secretUsername2, secretPassword2,
+		))
+
+	twoRegistriesSecretRegistryConfig := v1alpha1.Registry{
+		Server: secretHost1,
+		BasicAuth: v1alpha1.BasicAuthConfig{
+			SecretName:      twoRegistriesSecretName,
+			SecretNamespace: twoRegistriesSecretNamespace,
+		},
+	}
+
+	clientWithSingleRegistrySecret := testK8sClient.NewSimpleClientset(
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      singleRegistrySecretName,
+				Namespace: singleRegistrySecretNamespace,
+			},
+			Type: corev1.SecretTypeDockerConfigJson,
+			Data: map[string][]byte{
+				corev1.DockerConfigJsonKey: singleRegistrySecretContent,
+			},
+		},
+	)
+
+	clientWithTwoRegistriesSecret := testK8sClient.NewSimpleClientset(
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      twoRegistriesSecretName,
+				Namespace: twoRegistriesSecretNamespace,
+			},
+			Type: corev1.SecretTypeDockerConfigJson,
+			Data: map[string][]byte{
+				corev1.DockerConfigJsonKey: twoRegistriesSecretContent,
+			},
+		},
+	)
+
+	ctx := context.Background()
+	log := NewLogger()
+	job := Job{
+		log: log,
+	}
+
+	t.Run("bad_basic_auth_both_inline_and_secret", func(t *testing.T) {
```

Comment:
its not bad per-say.  The test is testing both inline and secret which is expected to fail.
This could be lifted out since it doesn't depend on anything above.

Reply:
It depends on the Job/ctx/logs etc. But I have fixed the name.